src/libdistgen.cpp

```diff
@@ -190,7 +233,9 @@ static void *thread_benchmark(void *arg) {
 			*ret += temp / (t2 - t1);
 		}
 	}
-
+#ifdef __QNXNTO__
+	free (rsizep);
```

Comment:
Okay, I see - we would need some kind of cleanup function to use the approach I suggest above. I assume `resized` is required for the whole lifetime of the thread.